common_types/src/schema.rs

```diff
@@ -151,18 +153,44 @@ pub enum Error {
 
     #[snafu(display("Arrow schema meta key not found.\nerr:\n{}", source))]
     ColumnSchemaDeserializeFailed { source: crate::column_schema::Error },
+
+    #[snafu(display("Failed to encode schema by protobuf, err:{}", source))]
+    EncodePbSchema {
+        source: protobuf::error::ProtobufError,
+    },
+
+    #[snafu(display("Encoded schema content is empty.\nBacktrace:\n{}", backtrace))]
+    EmptyEncodedSchema { backtrace: Backtrace },
+
+    #[snafu(display(
+        "Invalid schema encoding version, version:{}.\nBacktrace:\n{}",
+        version,
+        backtrace
+    ))]
+    InvalidSchemaEncodingVersion { version: u8, backtrace: Backtrace },
+
+    #[snafu(display(
+        "Failed to decode schema from protobuf bytes, buf:{:?}, err:{}",
+        buf,
+        source,
+    ))]
+    DecodeSchemaFromPb {
+        buf: Vec<u8>,
+        source: protobuf::error::ProtobufError,
+    },
 }
 
 pub type CatalogName = String;
 pub type SchemaId = u32;
 pub type SchemaName = String;
+pub type Result<T> = std::result::Result<T, Error>;
```

Comment:
Use `define_result!(Error)` to do this?

Reply:
`define_result!` is defined in `common_util` crate, which depends on `common_types`, so here we can't use this macro.